python/openinference-instrumentation/src/openinference/instrumentation/__init__.py

```diff
@@ -12,11 +31,150 @@ def suppress_tracing() -> Iterator[None]:
             # No tracing will occur within this block
             ...
     """
-    try:
-        from opentelemetry.context import _SUPPRESS_INSTRUMENTATION_KEY, attach, detach, set_value
-    except ImportError:
-        yield
+
+    def __enter__(self) -> "suppress_tracing":
+        self._token = attach(set_value(_SUPPRESS_INSTRUMENTATION_KEY, True))
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback) -> None:
+        """Raise any exception triggered within the runtime context."""
+        detach(self._token)
+        return None
+
+
+class UsingAttributes:
+    def __init__(
+        self,
+        session_id: str = "",
+        user_id: str = "",
+        metadata: Optional[Dict[str, str]] = None,
+        tags: Optional[List[str]] = None,
+    ) -> None:
+        self._session_id = session_id
+        self._user_id = user_id
+        self._metadata = metadata
+        self._tags = tags
+
+    def attach_context(self) -> None:
+        ctx = get_current()
+        if self._session_id:
+            ctx = set_value(SpanAttributes.SESSION_ID, self._session_id, ctx)
+        if self._user_id:
+            ctx = set_value(SpanAttributes.USER_ID, self._user_id, ctx)
+        if self._metadata:
+            ctx = set_value(SpanAttributes.METADATA, self._metadata, ctx)
+        if self._tags:
+            ctx = set_value(SpanAttributes.TAG_TAGS, self._tags, ctx)
+
+        self._token = attach(ctx)
+        return
+
+    def detach_context(self) -> None:
+        detach(self._token)
+        return
```

Comment:
Nit: Not sure these `return` statements are doing much for us.

Reply:
Removed